packages/ketcher-react/src/script/editor/Editor.ts

```diff
@@ -538,6 +557,289 @@ class Editor implements KetcherEditor {
+      selectedStruct.bonds.add(newBond);
+
+      attachmentPoints.set(attachmentAtomIdInSelectedStruct, newLeavingAtomId);
+      this.atomIdsMap.set(attachmentAtomIdInSelectedStruct, attachmentAtomId);
+    });
+
+    this._monomerCreationState = {
+      originalStruct: currentStruct.clone(),
+      attachmentPoints,
+    };
+
+    this.render.monomerCreationRenderState = {
+      attachmentPoints,
+    };
+
+    this.struct(selectedStruct);
+  }
+
+  closeMonomerCreationWizard() {
+    if (!this._monomerCreationState) {
+      return;
+    }
+
+    this.render.monomerCreationRenderState = null;
+    this.struct(this._monomerCreationState.originalStruct);
+    this._monomerCreationState = null;
+
+    this.tool('select');
+  }
+
+  saveNewMonomer(data) {
+    if (!this._monomerCreationState) {
+      throw new Error(
+        'Monomer creation wizard is not active, cannot save new monomer',
+      );
+    }
+
+    const ketSerializer = new KetSerializer();
+    const ketMicromolecule = JSON.parse(
+      ketSerializer.serialize(this.render.ctab.molecule),
+    );
+
+    const { symbol, name, type, naturalAnalogue } = data;
+
+    const attachmentPoints: IKetAttachmentPoint[] = [];
+    this._monomerCreationState.attachmentPoints.forEach(
+      (leavingAtomId, attachmentAtomId) => {
+        const attachmentPoint: IKetAttachmentPoint = {
+          attachmentAtom: attachmentAtomId,
+          leavingGroup: {
+            atoms: [leavingAtomId],
+          },
+          type:
+            attachmentPoints.length === 0
+              ? 'left'
+              : attachmentPoints.length === 1
+              ? 'right'
+              : 'side',
+        };
+        attachmentPoints.push(attachmentPoint);
+      },
+    );
+
+    const monomerTemplate: IKetMonomerTemplate = {
+      type: KetTemplateType.MONOMER_TEMPLATE,
+      id: `${symbol}___${name}`,
+      class: type,
+      classHELM:
+        type === KetMonomerClass.AminoAcid
+          ? MONOMER_CONST.PEPTIDE
+          : type === KetMonomerClass.CHEM
+          ? MONOMER_CONST.CHEM
+          : MONOMER_CONST.RNA,
+      alias: symbol,
+      fullName: name,
+      naturalAnalogShort: naturalAnalogue,
+      // TODO: Normalize atoms positions to avoid incorrect positioning upon expand/collapse
+      atoms: ketMicromolecule.mol0.atoms,
+      bonds: ketMicromolecule.mol0.bonds,
+      attachmentPoints,
+      root: {
+        nodes: [],
+        // TODO: Revisit IKetMonomerTemplate type
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        connections: [],
+        templates: [
+          {
+            $ref: `${KetTemplateType.MONOMER_TEMPLATE}-${symbol}___${name}`,
+          },
+        ],
+      },
+    };
+
+    const { root: templateRoot, ...templateData } = monomerTemplate;
+    const libraryItem = JSON.stringify({
+      root: {
+        ...templateRoot,
+      },
+      [`${KetTemplateType.MONOMER_TEMPLATE}-${symbol}___${name}`]: {
```

Comment:
Mb we can better use setMonomerTemplatePrefix()

Reply:
Fixed